source/tutorials/zfs.rst

```diff
@@ -0,0 +1,299 @@
+it's best that you get the source code from the ZFS on Linux repository,
+hosted on Github.
+
+Fetch the ZFS repository via git:
+
+.. code-block:: bash
+
+   git clone https://github.com/openzfs/zfs.git /opt/src/zfs
+
+Remember where you check-out the git repository, because you will need it
+when you upgrade ZFS. I have chosen /opt/src/zfs, but you can choose any
+workspace you like. ZFS will not run from this location -- we copy the
+source code into a DKMS directory in the next step.
+
+Do not delete the source location when you have completed this tutorial:
+you will need it later.
+
+Compile the module
+==================
+
+We will build the module using DKMS. This will keep the module up to date
+later as new kernels are released.
+
+You have already fetched the zfs codebase. Check out the tagged version
+that you plan to use. As of the time of this writing, the latest release
+tag is `0.8.4`:
+
+.. code-block:: bash
+
+    cd /opt/src/zfs
+    git checkout 0.8.4
+
+Copy the source code into `/usr/src/zfs-0.8.4`. This exposes the source
+code to DKMS. We will build the code from the new location:
+
+.. code-block:: bash
+
+    sudo cp -Rv /opt/src/zfs /usr/src/zfs-0.8.4
+    cd /usr/src/zfs-0.8.4
+
+The ZFS distribution provides a script to build a suitable dkms.conf file.
+Build dkms.conf and install it into the DKMS tree.
+
+.. code-block:: bash
+
+   cd /usr/src/zfs-0.8.4
+   scripts/dkms.mkconf -n zfs -v 0.8.4 -f dkms.conf
+   sudo dkms add -m zfs -v 0.8.4
+   sudo dkms build -m zfs -v 0.8.4
+   sudo dkms install -m zfs -v 0.8.4
+
+This will install the zfs kernel modules to:
+
+    /usr/lib/modules/<kernel-name>/extra/zfs
+
+This will install the zfs userspace tools to:
+::
+
+   + /usr/local/
+   |--+ bin/
+      |--zvol_wait
+      |--zgenhostid
+      |--raidz_test
+   |--+ etc/
+      |--+ zfs/
+         |--* zed.d/
+         |--+ zpool.d/
+         |--zfs-functions
+   |--+ include/libzfs/ [contents omitted]
+   |--+ lib/
+   |--+ libexec/
+      |--+ zfs/
+         |-- zpool.d/
+         |-- zed.d
+   |--+ share/zfs/ [contents omitted]
+   |--+ sbin/
+      |--fsck.zfs
+      |--zpool
+      |--zdb
+      |--zed
+      |--zfs
+      |--zhack
+      |--zinject
+      |--zpool
+      |--ztest
+      |--zstreamdump
+   |--+ src/
+      |--+ zfs-0.8.4/
+      |--+ spl-0.8.4/
+
+
+Load the new kernel module:
+
+.. code-block: bash
+
+   sudo modprobe zfs
+
+
+Set up systemd
+==============
+
+You should now have these unit files available:
+
+```
+zfs-env-bootfs.service
+zfs-zed.service
+zfs-import-cache.service
+zfs-import-scan.service
+zfs-mount.service
+zfs-share.service
+zfs-volume-wait.service
+```
+
+If you want to use ZFS automatic zpool import and filesystem
+mount services, copy the systemd.unit files into /etc and enable them:
+
+.. code-block:: bash
+
+   sudo cp ./etc/systemd/system/*.service /etc/systemd/system/
+   sudo cp ./contrib/dracut/90zfs/zfs-env-bootfs.service /etc/systemd/system/
```

Comment:
Might be better to symlink these?

Reply:
I updated to symlink just the two services we plan to use into /etc/systemd/system, and added a sentence explaining that the user should do the same for any other service units they plan to use.